### ISSUE TITLE
Publish Android TV client on Jio Store

### ISSUE THREAD

**Issue by harshitgoel96** [jellyfin/jellyfin-androidtv#3570]
### Describe the feature you'd like

Hi,
Jio Fibre is a leading ISP in India, that is providing a lot of content to Indian market. They have an Android STB that has its own play store. Can you please work to release the app on Jio store? f
The developer account and everything is free on jio platform.f

https://developer.jio.com/

**Comment by nielsvanvelzen** [comment]
We currently offer the app on F-Droid for devices without access to Google Play or the Amazon Appstore. It is also possible to sideload the apk file yourself. I'm not entirely against adding a new distribution channel, but it should have a good reason to exist.
I've never heard of Jio myself and their developer site doesn't provide any information about their product. This doesn't convince me to distribute our app on the Jio Store.

**Comment by PratikMahajan** [comment]
If you want a comparison, Jio handles more data than a combined [entity of Verizon+At&t+T-mobile in the US](https://timesofindia.indiatimes.com/technology/tech-news/reliance-jio-has-become-the-worlds-largest-mobile-operator-surpassing-china-mobile-with-this-new-record/articleshow/109550190.cms). 
Jio has an Android set-top box that is locked-down version of AndroidTV and there is no adb access to side-load any app on the set-top box. 
If anyone wants to watch Jellyfin on a Jio box, their only option today is to shift to some other OS.

**Comment by akshaykhairmode** [comment]
+1 for jellyfin on jio stb

**Comment by singhdilveer1** [comment]
Hello,

I'd like to help publish the Jellyfin app on the Jio Android Set Top Box platform on your behalf.

Jio is an Indian company and I think, as an Indian, I'll have an easier time registering and publishing the app on your behalf.

Regards.

**Comment by singhdilveer1** [comment]
Update - The Android TV client seems to have already been submitted by somebody so the dev portal refuses to accept the same package name again.

I wanted to ask if somebody on the dev team has already tried submitting the app on that store.

**Comment by nielsvanvelzen** [comment]
We are not currently publishing the app to the Jio Store. The app is available on Google Play, the Amazon Appstore, F-Droid, the GitHub releases on Thai repository and repo.jellyfin.org.

Please refrain from taking action on your own. Do not publish the app to other stores without our explicit permission.

**Comment by singhdilveer1** [comment]
> We are not currently publishing the app to the Jio Store. The app is available on Google Play, the Amazon Appstore, F-Droid, the GitHub releases on Thai repository and repo.jellyfin.org > 
> Please refrain from taking action on your own. Do not publish the app to other stores without our explicit permission.


Thank you for your response. I understand and respect your decision.

In light of this, would it be acceptable if I publish a fork of the Jellyfin TV app to the Jio Store? It'll be under a different name as to not cause you the headache of handling bug reports from a platform you don't support.

Jio's Android TV boxes don't support sideloading, and even the package manager has disappeared entirely after a recent update. 

Additionally, there are no developer options, tools, or documentation available that might indicate sideloading support for testing apps.
An unofficial app seems to be the only option for now.


Thank you.

**Comment by nielsvanvelzen** [comment]
Yes, if your fork adheres to our code license and does not use our branding (name/logo) it's fine.

**Comment by niravjdn** [comment]
Has it been published on jio?

**Comment by imshivlok** [comment]
Did you publish the fork? What's the name?

**Comment by singhdilveer1** [comment]
I'm not an app developer and I have a 9-5 job, so I don't have much time and there's a bit of a learning curve, even for something as basic as rebranding. To make matters worse, the Jio Developer Portal has been unreliable, and there's no way to directly contact their development team (they remotely enable debugging options on the set-top box). Despite the abundance of documentation and SDK resources—which are essentially just copied from VS Code and the Android website—they failed to provide any meaningful contact information. For example, here's a link to their guide about enabling debug options: [How to Push Your App on Jio STB.pdf](https://github.com/user-attachments/files/17991360/How.to.push.your.app.on.Jio.STB.pdf).  

Adding to the frustration, their helpline executives have no idea if a dedicated app development support team even exists within Jio. To top it off, they've classified some documents as "Confidential" but released them on the public portal without requiring a login. The whole system is a mess. 

So yeah, if you leave it up to me, It's gonna take a while.

**Comment by imshivlok** [comment]
Yeah, truly. It is a mess. The JioSTB actually has pretty decent potential. I'd love to emulate retro consoles like the N64, GBA/GBC, Nintendo, PSPs, and more. I'm pretty sure it would be easy, only if there was a package manager, which they apparently removed by their latest update.
I originally got the STB thru an AirFiber connection, and oh gosh! I hate both of them. The WiFi needs a restart everyday to provide a stable internet connection. The Router Configs are all messed up, like you can't even split the 2.4/5Ghz bands in it. It comes with a single combined AP for both the bands. There's no option for port forwarding, there is also no way to access the USB drive (if you plan to) plugged at the I/O. It also does not support bridge mode or anything like that. It's a simple, pay and internet system. The speeds are also unstable, poor pings while gaming. Troubles connecting a power backup with the router because apparently it comes with a special "power brick", so you'll need a UPS in any case of power cuts.
Additionally, 707 rupees for a 30Mbps connection is just so unfair. There are broadbands like excitel and netplus (apparently not available at my location) providing 100- even 200 Mbps at 499/599rs.
The STB is just rendered useless, as my TV apparently has a lot more polished software than this piece of crap. It has random youtube buffering issues too.
I don't have a reason for further continuing this service, apart from the fact that wired broadbands are not available at my locality and cellular connections aren't really great, so I'll just have to stick with it for now, but as soon as a fiber becomes available nearby. I'm going to dump this piece of crap! TERRIBE EXPERIENCE WITH IT.

**Comment by imshivlok** [comment]
Oh, and it doesn't have audio out btw. HDMI is the only way.

**Comment by IsotoxalDev** [comment]
I use jellyfin a lot to stream music. I want to play the same through my Jio set-top box (Android TV). I can't neither access google play store or amazon store or f-droid in this set-top box. The only option is JIO store. The only option remains is to flash a custom rom which is very risky and not many people have tried. Having this app available in jiostore would be really appreciated.

**Comment by rndnanthu** [comment]
> We've decided to not release our app to the Jio Store in any official way for the foreseeable future, as such I'll be closing this issue. As mentioned before, you're free to publish a rebranded fork of the app yourself.

That good decision. JioSTB is useless device with full of hindutuva propaganda blotwares.

**Comment by IsotoxalDev** [comment]
> That good decision. JioSTB is useless device with full [...] blotwares. 

Woah.. I understand the bloatware part. but any religion or propaganda being a criteria is a shame. You have to consider that jio has brought internet connectivity to millions of people. I am not advocating for jio. just stating any religion or  propaganda should not seep into any open source project or any project for that matter. I have a jio stb at home i want to play songs from my jellyfin server. but i have no official means that would allow me to do so. even the jailbreak loopholes found are not there in the latest stb. idk exactly why but the browser in the stb also couldn't play the songs properly so an official release is appreciated by me

**Comment by thesanju** [comment]
Hello @singhdilveer1 @harshitgoel96 
I recently got a smart TV and jio STB and the smart TV also has jio tele OS, this makes it impossible to download jellyfin in any way, Can you help me guys?

**Comment by imshivlok** [comment]
there's no other way but jellyfin web (i got rid of jiostb altogether. it was shit)
